charts/argocd-apps/templates/applications.yaml

```diff
@@ -49,3 +49,19 @@ spec:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 {{- end }}
+
+{{- range .Values.itemTemplates }}
+{{- if kindIs "string" .template }}
+{{- $template := .template -}}
+{{- range .items }}
+---
+{{ tpl $template (set . "Template" $.Template) }}
+{{- end }}
+{{- else }}
+{{- $template := .template | toYaml -}}
+{{- range .items }}
+---
+{{ tpl $template (set . "Template" $.Template) }}
+{{- end }}
+{{- end }}
+{{- end }}
```

Comment:
Can you move this block to a dedicated file. Doesn't make sense to keep this inside applications.yaml.

Please create a new file for that. Eg. `item-templates.yaml`.